### PR TITLE
Issue 1645 investigate black

### PR DESCRIPTION
**Pull request by emdv88**

### CODE REVIEW

**Comment by emdv88** [comment]
Still need to take redo the black formatting after a rebase to prevent merge conflicts

also still have to include black in the developer manual

so not yet ready for merge

**Comment by emdv88** [comment]
Should be ready for merge now, except some newly introduced issues with sonarcloud, deepsource ant better code hub.
As far as I have checked they were all pre-existing issues that have now been picked up due to the core reformatting.

**Comment by emdv88** [comment]
now ready for a step by step introduction of black formatting. The git hook will only format python files that have been changed.

File: docs/DEVELOP.md [hunk 1]
```diff
@@ -100,6 +100,8 @@ This section contains some notes on coding style used in this project. It's far
 
 Most of the coding standard are enforced by the [quality checks](#quality-checks).
 
+To automate autoformatting of python code using black, a git pre-commit hook is provided at `git-hooks/pre-commit`. To use it and automatically formal all changed python files on committing, copy this script into your local `.git/hooks` folder of this repo.
```
Comment (fniessink) [inline docs/DEVELOP.md]:
```suggestion
To enable autoformatting of Python code using black, a git pre-commit hook is provided at `git-hooks/pre-commit`. To use it and automatically format all changed Python files on committing, copy this script into your local `.git/hooks` folder of this repo.
```

**Review by fniessink — CHANGES_REQUESTED** [review]

**Pull request merged**